src/workerd/io/worker.c++

```diff
@@ -1971,7 +1971,15 @@ kj::Maybe<kj::Own<api::ExportedHandler>> Worker::Lock::getExportedHandler(
 
   kj::StringPtr n = name.orDefault("default"_kj);
   KJ_IF_SOME(h, worker.impl->namedHandlers.find(n)) {
-    return fakeOwn(h);
+    kj::Own<api::ExportedHandler> ret = fakeOwn(h);
+    jsg::Lock& js = *this;
+    if (FeatureFlags::get(js).getCtxPropsToDefault()) {
+      api::ExportedHandler constructedHandler = h.clone(js);
+      constructedHandler.ctx = jsg::alloc<api::ExecutionContext>(js, props.toJs(js));
+      kj::Own<api::ExportedHandler> handlerPtr = fakeOwn(constructedHandler);
```

Comment:
@kentonv doing this feels off, how do I get a proper allocation for an ExportedHandler?

Reply:
As written this results in use-after-free -- definitely bad.

You can replace `fakeOwn(constructedHandler)` with `kj::heap(mv(constructedHandler))` to move it to a heap-allocated object.